File: fluentcore/src/main/java/com/mcreater/fxfluent/xaml/tags/SolidColorBrushContentTag.java
package com.mcreater.fxfluent.xaml.tags;

import com.mcreater.fxfluent.brush.SolidColorBrush;
import com.mcreater.fxfluent.xaml.XamlHelper;

public class SolidColorBrushContentTag extends SimpleContentTag<SolidColorBrush> {
    public SolidColorBrush toObject() {
        String content = getElement().attributeValue("Color");
        if (content == null) return null;
        String opacity = getElement().attributeValue("Opacity");
        double opacityD = opacity == null ? 1.0 : Double.parseDouble(opacity);

        return new SolidColorBrush(XamlHelper.parseAnyColor(dict, content, opacityD));
    }
}

File: fluentcore/src/main/java/com/mcreater/fxfluent/xaml/tags/SolidColorBrushContentTag.kt
package com.mcreater.fxfluent.xaml.tags

import com.mcreater.fxfluent.brush.SolidColorBrush
import com.mcreater.fxfluent.xaml.XamlHelper.Companion.parseAnyColor
import javafx.scene.paint.Color


class SolidColorBrushContentTag : SimpleContentTag<SolidColorBrush?>() {
    override fun toObject(): SolidColorBrush {
        val content = element!!.attributeValue("Color") ?: return SolidColorBrush(Color.TRANSPARENT)
        val opacity = element!!.attributeValue("Opacity")
        val opacityD = opacity?.toDouble() ?: 1.0
        return SolidColorBrush(parseAnyColor(dict, content, opacityD))
    }
}

